algorithims/a_star/fieldGraph.py

```diff
@@ -0,0 +1,40 @@
+class Node:
+    def __init__(self, position):
+        self.previous = None
+        self.position = position
+        self.g = 0 # distance so far
+        self.h = 0 # heuristic/distance to target
+        self.f = 0 # g plus h
+        self.neighbours = []
+
+    def __lt__(self, other):
+        if (self.f == other.f):
+            return self.h < other.h
+        return self.f < other.f
+
+    def __gt__(self, other):
+        return self.f > other.f
+
+
+# creates nodes on the field and add them to the graph that astar is going to use
+# currently only being used for testing the astar algorithm
+class FieldGraph():
```

Comment:
@Ana-Monici como eu implementaria aqui se tivesse uma heusristica externa? a ideia eh que usemos ```FieldGraph``` como uma classe abstrata ou exemplo certo?

Se for assim, recomendo que esse exemplo esteja em outro lugar (pode ser numa documentacao na wiki, por exemplo), por que ele nao faz parte do funcionamento do sistema. Tambem, podemos dar o exemplo ja implementando num modulo de strategy, como foi feito [aqui para os campos potencias](https://github.com/project-neon/NeonFC/blob/master/strategy/tests/pfScratch.py).